aten/src/ATen/TensorIndexing.h

```diff
@@ -385,7 +397,7 @@ static inline Tensor handleDimInMultiDimIndexing(
       index.slice().step(),
       /*disable_slice_optimization=*/disable_slice_optimization,
       original_tensor_device,
-      prev_dim_result_sizes);
+      *prev_dim_result_sizes);
```

Comment:
I think it might be worthwhile adding a `TORCH_INTERNAL_ASSERT_DEBUG_ONLY(prev_dim_result_sizes)` in case this assumption ever does not hold. I'd also annotate that assert with some comment that explains why we can always assume `prev_dim_result_sizes` has a value in this case.

Reply:
It actually can be empty: when using a tuple to index a nested tensor, will call `applySlicing` -> repeat `handleDimInMultiDimIndexing` for all indices, which means at the first pass it can be handling a nested tensor who has empty `prev_dim_result_sizes`

Reply:
but for latter passes it must have a value, though, since it would always be normal tensor by then

Reply:
So does that mean that adding this check would be redundant or that `prev_dim_result_sizes` always has a value, but might be an empty IntArrayRef? If it always has a value, does this need to be an optional IntArrayRef or can it just be a regular IntArrayRef that happens to be empty sometimes?

Reply:
I should be more precise: `prev_dim_result_sizes` is `c10::nullopt` in the first call, then regular `IntArrayRef` in the latter calls

Reply:
so we cannot have the `TORCH_INTERNAL_ASSERT_DEBUG_ONLY(prev_dim_result_sizes)`: it does not hold in the first call

Reply:
But it has to hold true in this particular branch, otherwise `* prev_dim_result_sizes` should [violate an assertion](https://github.com/pytorch/pytorch/blob/master/c10/util/Optional.h#L739-L751).

Reply:
@cpuhrsch the updated version of the code no longer unconditionally dereferences `prev_dim_result_sizes` for this branch